forest/cli/src/cli/wallet_cmd.rs

```diff
@@ -241,3 +274,140 @@ impl WalletCommands {
         }
     }
 }
+
+fn format_balance_string(num: Decimal, mode: FormattingMode) -> String {
+    let units = ["atto", "femto", "pico", "nano", "micro", "milli", ""];
+    let orig = num;
+
+    let mut num = num;
+    let mut unit_index = 0;
+    while num >= dec!(1000.0) && unit_index < units.len() - 1 {
+        num /= dec!(1000.0);
+        unit_index += 1;
+    }
+
+    match mode {
+        FormattingMode::ExactFixed => {
+            let fil = orig / dec!(1e18);
+            format!("{fil} FIL")
+        }
+        FormattingMode::NotExactFixed => {
+            let fil_orig = orig / dec!(1e18);
+            let fil = fil_orig.round_dp_with_strategy(
+                NUM_SIGNIFICANT_DIGITS,
+                RoundingStrategy::MidpointAwayFromZero,
+            );
+            let mut res = format!("{} FIL", fil.trunc());
+            if fil != fil_orig {
+                res.insert(0, '~');
+            }
+            res
+        }
+        FormattingMode::ExactNotFixed => format!("{num:0} {} FIL", units[unit_index]),
+        FormattingMode::NotExactNotFixed => {
+            let mut fil = num.round_dp_with_strategy(
```

Comment:
Think we want `round_sf_with_strategy` here. Significant digits = `fs`, decimal points = `dp`.

Reply:
Tested this function out and it forces 4 digits, even with trailing zeros, so I reverted it to former state, since it worked perfectly well